python/hidet/graph/ops/definitions/conv3d_transpose/conv3d_transpose.py

```diff
@@ -0,0 +1,112 @@
+        h = (p - 1) * sx + -px0 - px1 + kx + output_padding[1]
+        w = (q - 1) * sy + -py0 - py1 + ky + output_padding[2]
+
+        if output_padding[0] >= stride[0] or output_padding[1] >= stride[1] or output_padding[2] >= stride[2]:
+            raise ValueError(
+                'Conv3dTranspose expect the output_padding < stride, \n'
+                f'but got output_padding, stride: {output_padding}, {stride}'
+            )
+        if any(p < 0 for p in padding):
+            raise ValueError('Negative padding is not supported.')
+
+        og = oc // groups  # output channels in each group
+        output = compute(
+            name='out',
+            shape=[n, c, z, h, w],
+            fcompute=lambda ni, ci, zi, hi, wi: reduce(
+                shape=[og, kz, kx, ky],
+                fcompute=lambda ogi, kzi, kxi, kyi: if_then_else(
+                    cond=LogicalAnd.join(
+                        zi + pz0 >= kzi,
+                        zi + pz0 < r * sz + kzi,
+                        (zi + pz0 - kzi) % sz == 0,
+                        hi + px0 >= kxi,
+                        hi + px0 < p * sx + kxi,
+                        (hi + px0 - kxi) % sx == 0,
+                        wi + py0 >= kyi,
+                        wi + py0 < q * sy + kyi,
+                        (wi + py0 - kyi) % sy == 0,
+                    ),
+                    then_expr=(
+                        data[
+                            ni,
+                            (ci // wc) * og + ogi,
+                            (zi + pz0 - kzi) // sz,
+                            (hi + px0 - kxi) // sx,
+                            (wi + py0 - kyi) // sy,
+                        ]
+                        * weight[(ci // wc) * og + ogi, ci % wc, kzi, kxi, kyi]
+                    ),
+                    else_expr=0.0,
+                ),
+                reduce_type='sum',
+            ),
+        )
+        super().__init__(name='conv3d_transpose', inputs=[data, weight], outputs=[output])
+
+
+class Conv3dTransposeOp(Operator):
+    def __init__(
+        self,
+        x: Tensor,
+        w: Tensor,
+        stride: Tuple[int, int, int],
+        padding: Tuple[int, int, int, int, int, int],
+        groups: int,
+        output_padding: Tuple[int, int, int],
+    ):
+        super().__init__(
+            inputs=[x, w],
+            task=Conv3dTransposeTask(input_like(x, 'x'), input_like(w, 'w'), stride, padding, groups, output_padding),
+            attributes={'stride': stride, 'padding': padding, 'groups': groups, 'output_padding': output_padding},
+        )
+
+
+def conv3d_transpose(
+    data: Tensor,
+    weight: Tensor,
+    stride: Union[int, Sequence[int]],
+    padding: Union[int, Sequence[int]],
+    groups: int = 1,
+    output_padding: Union[int, Sequence[int]] = 0,
+) -> Tensor:
+    sz, sx, sy = normalize_stride(stride, dim=3)
+    pz0, px0, py0, pz1, px1, py1 = normalize_padding(padding, dim=3)
```

Comment:
Any reason to use the order of `z, x, y` instead of `x, y, z`?

Reply:
Hey @yaoyaoding , I was just following the structure according to PyTorch where they put depth first
input: (N, C, D, H, W) 
D is z in this case

Reply:
Make sense. Thanks.